kolibri/core/assets/src/validators.js

```diff
@@ -25,3 +25,195 @@ export function validateLearningActivity(arr) {
   const isValidLearningActivity = v => Object.values(LearningActivities).includes(v);
   return arr.length > 0 && arr.every(isValidLearningActivity);
 }
+export function objectPropValidator( options ){
```

Comment:
I think overall this could be simplified and cleaned up with a type map:
```
import isNumber from 'lodash/isNumber';
import isBoolean from 'lodash/isBoolean';

const typeValidatorMap = {
  Number: isNumber,
  Boolean: isBoolean,
}
```
This is an example, but could be extended to all supported types.

Then when going to check can just do:
```
typeValidatorMap[option.type](data);
```